scripts/doc_checker.py

```diff
@@ -307,8 +308,15 @@ def get_noqa_checks(doc: Docstring) -> list:
                 noqa_str = line
                 break
     else:
-        # pydocstyle only check first line; aling with it
-        noqa_str = source.split("\n", 1)[0]
+        # noqa string is defined as the first line before the docstring
+        if not doc.raw_doc:
+            # noqa string is meaningless if there is no docstring in module
+            return []
```

Comment:
what if I want to disable a check for the whole module (not a check for module docstring)? don't we propagate module-level `noqa` to function level?

Reply:
This will change the behavior of `numpydoc` quite a lot. Well, and immediately the first question regarding the implementation - how to distinguish between situations when we need to disable checks only for the module docstring and when we need to disable checks for the entire file. This adds another layer of complexity...